singer/src/main/java/com/pinterest/singer/config/DirectorySingerConfigurator.java

```diff
@@ -114,13 +116,18 @@ public SingerConfig parseSingerConfig() throws ConfigurationException {
           singerLogConfig = LogConfigUtils.parseLogConfigFromFile(newFile);
         }
         singerConfig.addToLogConfigs(singerLogConfig);
-      } catch (ConfigurationException e) {
-        Stats.incr(SingerMetrics.SINGER_CONFIGURATOR_CONFIG_ERRORS);
-        LOG.error("Failed to parse Singer client config file {}, exception: {}. Skip and continue.",
+      } catch (Exception e) {
+        LOG.error("Failed to parse log config file {}, exception: {}, Skip and continue.",
             newFile.getPath(), ExceptionUtils.getFullStackTrace(e));
+        badConfigs++;
       }
     }
 
+    if (badConfigs > 0) {
+      Stats.setGauge(SingerMetrics.SINGER_CONFIGURATOR_CONFIG_ERRORS, badConfigs);
+      LOG.error("Number of bad log config files is {}", badConfigs);
```

Comment:
Is there an easy way to have a specifier in the metric to simply tells us which config has the issue? Or we have to refer to the logs?

Reply:
Yes, we can use OpenTSDBMetricConverter but I just didn't use it to prevent high tag cardinality. Though the chances of that are pretty minimal I suppose

Reply:
Yes, that's what I thought too. It can be a future improvement though. Up to you.